### PR TITLE
Fix ffmpeg static libraries link against libx264.dll.a & libxml2.dll.a  &  libopenjp2.dll.a

### PR DESCRIPTION
**Pull request by shithappens2016**
avoid ffmpeg static libraries link against libx264.dll.a & libxml2.dll.a & libopenjp2.dll.a
#8824

### CODE REVIEW

File: mingw-w64-ffmpeg/PKGBUILD [hunk 1]
```diff
@@ -122,8 +126,12 @@ build() {
     mkdir -p "${srcdir}/build-${MINGW_CHOST}${_variant}" && cd "${srcdir}/build-${MINGW_CHOST}${_variant}"
     if [[ ${_variant} == -static ]]; then
       ENABLE_VARIANT=--enable-static
+      CFLAGS="$CFLAGS_BK -DLIBXML_STATIC -static -static-libgcc"
+      CXXFLAGS="$CXXFLAGS_BK -DLIBXML_STATIC -static -static-libgcc -static-libstdc++"
```
Comment (lazka) [inline mingw-w64-ffmpeg/PKGBUILD:130]: Why is this needed?

Comment (shithappens2016) [reply mingw-w64-ffmpeg/PKGBUILD:130]: You mentioned that
> It's a feature of `pkgconf`, if you add `Cflags.private` it is returned instead of `Cflags` if `--static` is used, just like with `Libs` and `Libs.private`

**Comment by shithappens2016** [comment]
I  have made the change.

File: mingw-w64-ffmpeg/PKGBUILD (hunk 1 above)
Comment (shithappens2016) [reply mingw-w64-ffmpeg/PKGBUILD]:
```shell
$ nm libavformat.a |grep xml
0000000000000000 r .rdata$.refptr.xmlFree
0000000000000000 R .refptr.xmlFree
                 U xmlCheckVersion
                 U xmlCleanupParser
                 U xmlCopyNode
                 U xmlDocGetRootElement
                 U xmlFirstElementChild
                 U xmlFree
                 U xmlFreeDoc
                 U xmlFreeNode
                 U xmlGetProp
                 U xmlNewNode
                 U xmlNextElementSibling
                 U xmlNodeGetContent
                 U xmlNodeSetContent
                 U xmlReadMemory

```
I have made the change, static libraries is link against libxml2.a correctly.

but there's another problem.

`pkg-config -cflags --static`  will return **Cflags + Cflags.private** not just Cflags.private.

seems `-DX264_API_IMPORTS` had to be removed form Cflags.

Comment (shithappens2016) [reply mingw-w64-ffmpeg/PKGBUILD]: https://github.com/msys2/MINGW-packages/blob/985767f5f8dc64f35713090317a6d7a347ddb50d/mingw-w64-ffmpeg/PKGBUILD#L127-L135

static libraries are link against libx264.a correctly.

```shell
$ nm libavcodec.a |grep x264
                 U ff_libx264_encoder
                 U ff_libx264rgb_encoder
libx264.o:
0000000000000000 r .rdata$.refptr.x264_levels
0000000000000000 R .refptr.x264_levels
0000000000000100 D ff_libx264_encoder
0000000000000000 D ff_libx264rgb_encoder
0000000000000580 r x264_class
00000000000006a0 r x264_defaults
                 U x264_encoder_close
                 U x264_encoder_delayed_frames
                 U x264_encoder_encode
                 U x264_encoder_headers
                 U x264_encoder_maximum_delayed_frames
                 U x264_encoder_open_161
                 U x264_encoder_reconfig
                 U x264_levels
                 U x264_param_apply_fastfirstpass
                 U x264_param_apply_profile
                 U x264_param_cleanup
                 U x264_param_default
                 U x264_param_default_preset
                 U x264_param_parse
                 U x264_picture_init
0000000000002680 r x264_preset_names
0000000000002560 r x264_profile_names
00000000000025e0 r x264_tune_names
```

**Comment by shithappens2016** [comment]
https://github.com/msys2/MINGW-packages/blob/985767f5f8dc64f35713090317a6d7a347ddb50d/mingw-w64-ffmpeg/PKGBUILD#L127-L135

static libraries are link against libx264.a correctly.

```shell
$ nm libavcodec.a |grep x264
                 U ff_libx264_encoder
                 U ff_libx264rgb_encoder
libx264.o:
0000000000000000 r .rdata$.refptr.x264_levels
0000000000000000 R .refptr.x264_levels
0000000000000100 D ff_libx264_encoder
0000000000000000 D ff_libx264rgb_encoder
0000000000000580 r x264_class
00000000000006a0 r x264_defaults
                 U x264_encoder_close
                 U x264_encoder_delayed_frames
                 U x264_encoder_encode
                 U x264_encoder_headers
                 U x264_encoder_maximum_delayed_frames
                 U x264_encoder_open_161
                 U x264_encoder_reconfig
                 U x264_levels
                 U x264_param_apply_fastfirstpass
                 U x264_param_apply_profile
                 U x264_param_cleanup
                 U x264_param_default
                 U x264_param_default_preset
                 U x264_param_parse
                 U x264_picture_init
0000000000002680 r x264_preset_names
0000000000002560 r x264_profile_names
00000000000025e0 r x264_tune_names
```

**Comment by shithappens2016** [comment]
All fixed

Except libgnutls(relies on libp11-kit not support static build) & libvulkan(static build not support)

#8840
#8839

please review and merge @lazka
